addons/mass_mailing/models/mass_mailing.py

```diff
@@ -54,6 +54,50 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def copy(self, default=None):
+        recipients = self.env['mail.mass_mailing.contact'].search([('list_ids', 'in', self.ids), ('opt_out', '!=', True)])
+        mailing_list = super(MassMailingList, self).copy(default=default)
+        for recipient in recipients:
+            recipient.write({'list_ids': [(4, mailing_list.ids)]})
+        return mailing_list
+
+    def merge_massmail_list(self, mailing_lists):
```

Comment:
@tde-banana-odoo ,
i think we did same thing what you said.
problem is when we update unique contacts in destination mailing list , we have to identify 
unique contact by his email id field . other wise every time it will add new contacts.

so we first fetch all distinct contact by email then write it on that contact with current destination mailing list.